data/core/highlighter.lua

```diff
@@ -49,7 +49,14 @@ function highlighter.tokenize(syntax, text, state)
       local s, e = find_non_escaped(text, p.pattern[2], i, p.pattern[3])
 
       if s then
-        push_token(res, p.type, text:sub(i, e))
```

Comment:
Maybe a better solution is just:

  push_token(res, p.type, text:sub(i, e):gsub("(.-)\n", "%1"))

or something similar, but I think that the regex-based string replacement is too heavy for just trimming the '\n' from the end of the token.